packages/cardano-services/src/Program/programs/pgBossWorker.ts

```diff
@@ -50,5 +51,15 @@ export const loadPgBossWorker = async (args: PgBossWorkerArgs, deps: LoadPgBossW
 
   if (!db) throw new MissingProgramOption(pgBossWorker, PostgresOptionDescriptions.ConnectionString);
 
+  logger.error('args: ', args);
```

Comment:
Apart from this line which seems to be a debug log line...

This is the right place to perform these kind of checks, but for arguments which have an impact on all the handlers; these checks have impact only on the metadata handler.

Performing these two checks in ` stakePoolMetadataHandlerFactory` between the two lines:

```
  const service = createHttpStakePoolMetadataService(logger);

  return async (task: StakePoolMetadataJob) => {
```

we can be sure that the condition `args.queues.includes(STAKE_POOL_METADATA_QUEUE)` is satisfied (otherwise the function is neither executed).

I suggest to move the two check to:

1. simplify the conditions inside the `if`s (removing the condition satisfied for sure if the flow passes from there)
2. aggregate _arguments checks_ and _init_ phases of the metadata handler inside the dedicated factory _function_.

Reply:
This is the line I added in our session to debug the argument problem. I removed it.  Also I agree with your suggestions.